syntax = "proto3";

option java_multiple_files = true;

package service_v2;

service Inference {
    rpc Inference (InferenceRequest) returns (InferenceReply) {
    }
}

message InferenceRequest {
    int32 photo_id = 1;
    int32 product_id = 2;
    int32 channel_id = 3;
    int32 width = 4;
    int32 height = 5;
    bytes encoded_image = 6;
    int32 x_offset = 7;
    int32 y_offset = 8;
}
message Point {
    int32 x = 1;
    int32 y = 2;
}
message SingleInferenceReply {
    repeated Point points = 1;
    string class_name = 2;
    float xmin = 3;
    float ymin = 4;
    float bb_width = 5;
    float bb_height = 6;
    float score = 7;
    float length = 8;
    float width = 9;
    float pixel_area = 10;
    float gradients = 11;
    float contrast = 12;
    float brightness = 13;
    float max20brightness = 14;
    float min20brightness = 15;
    bytes mask=16;
    int32 mask_width=17;
    int32 mask_height=18;
}

message InferenceReply {
    int32 num_detections = 1;
    int32 photo_id = 2;
    int32 product_id = 3;
    int32 channel_id = 4;
    repeated SingleInferenceReply singleInferenceReply = 5;
}
